### ISSUE TITLE
Keeping v1 commands available in v2 may confuse users

### ISSUE THREAD

**Issue by pzhlkj6612** [miurahr/aqtinstall#413]
## Overview

In [Legacy subcommands § Command Line Options — aqtinstall 2.0.0 documentation](https://aqtinstall.readthedocs.io/en/v2.0.0/cli.html#legacy-subcommands), we wrote:

> The subcommands `install`, `tool`, `src`, `doc`, and `examples` have been deprecated in favor of the newer `install-*` commands, but they remain in aqt in case you still need to use them.

After reading the above paragraph, I will think *"Fine, 'aqt tool' is still there. Those guys are keeping two series of commands in v2. I don't have to update my script after upgrading aqtinstall"*. However, the reality is that after switching to v2, the old script will **NOT** work without modification, at least for the "tool" command.

## Test

A working example for installing QIFW by aqtinstall 1.2.5:

```console
$ aqt tool linux tools_ifw 4.1.1 qt.tools.ifw.41
Specified target combination is not valid: linux tools_ifw qt.tools.ifw.41
...
Time elapsed: 8.37262754 second
```

The same script does not work in 2.0.0:

```console
$ aqt tool linux tools_ifw 4.1.1 qt.tools.ifw.41
usage: aqt tool ...
...
aqt tool: error: argument target: invalid choice: 'tools_ifw' (choose from 'desktop', 'winrt', 'android', 'ios')
Error: Process completed with exit code 2.
```

Rewrite the script according to the documentation of "[install-tool](https://aqtinstall.readthedocs.io/en/v2.0.0/cli.html#install-tool-command)" but keep the *legacy* "tool" command. It works in 2.0.0:

```diff
- aqt tool linux         tools_ifw 4.1.1 qt.tools.ifw.41
+ aqt tool linux desktop tools_ifw
```

```console
$ aqt tool linux desktop tools_ifw
aqtinstall(aqt) v2.0.0 on Python 3.8.10 [CPython GCC 9.4.0]
Warning: The command 'tool' is deprecated and marked for removal in a future version of aqt.
In the future, please use the command 'install-tool' instead.
...
Time elapsed: 11.00345962 second
```

Add the "tool variant name":

```diff
- aqt tool linux desktop tools_ifw
+ aqt tool linux desktop tools_ifw qt.tools.ifw.41
```

```console
$ aqt tool linux desktop tools_ifw qt.tools.ifw.41
aqtinstall(aqt) v2.0.0 on Python 3.8.10 [CPython GCC 9.4.0]
Warning: The command 'tool' is deprecated and marked for removal in a future version of aqt.
In the future, please use the command 'install-tool' instead.
...
Time elapsed: 9.82558412 second
```

*Use the command 'install-tool' instead*:

```diff
- aqt tool         linux desktop tools_ifw qt.tools.ifw.41
+ aqt install-tool linux desktop tools_ifw qt.tools.ifw.41
```

```console
$ aqt install-tool linux desktop tools_ifw qt.tools.ifw.41
aqtinstall(aqt) v2.0.0 on Python 3.8.10 [CPython GCC 9.4.0]
...
Time elapsed: 11.23552921 second
```

## Conclusion

As you can see, in aqtinstall v2, "tool" is not a legacy subcommand but **another name** of "install-tool". Those old interfaces have been changed, so keeping them available in v2 may confuse users.

I really know that the API changes can be (or should be) made in a new major release, but we didn't keep the availability and compatibility for those legacy commands, and what's worse is that we didn't document it correctly and clearly.

## Ignorable possible solutions

Here are some expensive ways to solve this problem:

- Remove those legacy commands in the next major version and release this new version as soon as possible. Yes, aqtinstall v3.
- Or, make those legacy commands usable again in the next minor version of v2. In other words, keep two series (v1 and v2) of commands in aqtinstall ^v2.1.

**Comment by ddalcino** [comment]
Good catch, thank you @pzhlkj6612! I think I was responsible for this when I wrote #355. In that PR, I implemented the backwards compatibility layer: it’s just an additional command line interface that maps the old syntax to the new commands. I clearly messed up on `aqt tool`.

I consider this to be a bug. I would prefer to release the bug fix as 2.0.1, because it would fix something that 2.0.0 claims to do properly, but does not. This would be fixing an existing broken interface, rather than introducing a new one.

It’s possible that this kind of bug fix cannot be released as 2.0.1 under semantic versioning rules; in that case we can have a super early 2.1.0?

In either case, I think we should audit the test code to make sure that all commands are being tested adequately.

**Comment by pzhlkj6612** [comment]
Hi, @ddalcino . Thank you for the rapid response.

<br>

> I consider this to be a bug... This would be fixing an existing broken interface, rather than introducing a new one.

Well, this is indeed a bug. Before getting your response, I thought that supporting v1 commands is an unfinished implementation or removal in a new major release, so I gave some radical solutions.

> ...  I would prefer to release the bug fix as 2.0.1, because it would fix something that 2.0.0 claims to do properly, but does not...
> 
> It’s possible that this kind of bug fix cannot be released as 2.0.1 under semantic versioning rules; in that case we can have a super early 2.1.0?

This is also what I concerned before posting this issue. Maybe we could mark the version 2.0.0 as "[yanked](https://pypi.org/help/#yanked)" after releasing 2.0.1? This will allow pip to ignore the defective release and warn the user who wants to use it (see [Installers § PEP 592](https://www.python.org/dev/peps/pep-0592/#installers)).

Hi, @miurahr . What do you think about this?

<br>

> In either case, I think we should audit the test code to make sure that all commands are being tested adequately.

I agree. Both correct and incorrect usage need to be tested.

<br>

> ... **Edit**: According to my tests in #414, I got `aqt examples` and `aqt src` wrong too.

Uh-oh.
